app/components/AuthForm.tsx

```diff
@@ -0,0 +1,28 @@
+"use client"
+
+type AuthFormProps = {
+  onSubmit: (formData: FormData) => Promise<void>
+  title: string
+  action: string
+}
+
+export const AuthForm: React.FC<AuthFormProps> = ({
+  onSubmit,
+  title,
+  action,
+}) => {
+  return (
+    <div className="flex min-h-screen flex-col items-center p-16">
+      <h1>{title}</h1>
+      <form action={onSubmit}>
+        <p>username</p>
+        <input placeholder="username" name="username"></input>
```

Comment:
it might be good to set `required` property on `username` and `password` fields for some better ux (front-end validation)